crypto/crypto.go

```diff
@@ -445,39 +450,74 @@ func VerifyLogicSig(lsig types.LogicSig, sender types.Address) (result bool) {
 		return false
 	}
 
-	result = false
 	toBeSigned := programToSign(lsig.Logic)
-	// logic sig, compare hashes
-	if !hasSig && !hasMsig {
-		result = types.Digest(sha512.Sum512_256(toBeSigned)) == types.Digest(sender)
-		return
-	}
 
 	if hasSig {
-		result = ed25519.Verify(sender[:], toBeSigned, lsig.Sig[:])
-		return
+		return ed25519.Verify(lsig.SigKey[:], toBeSigned, lsig.Sig[:])
 	}
 
-	result = VerifyMultisig(sender, toBeSigned, lsig.Msig)
```

Comment:
does this old code assumes `sender == msigAccount`` ?

Reply:
Yes it does. I believe this was written before rekeying.